filament/include/filament/Texture.h

```diff
@@ -261,6 +264,26 @@ class UTILS_PUBLIC Texture : public FilamentAPI {
          */
         Builder& external() noexcept;
 
+        /**
+         * Specifies a callback that will execute once the resource's data has been fully allocated
+         * within the GPU memory. This enables the resource creation process to be handled
+         * asynchronously.
```

Comment:
We should document what happens if this `Texture` (and likewise for `VertexBuffer`, `IndexBuffer`, etc) is used before the async creation process has finished. Is it an error?